src/Model/Model.ts

```diff
@@ -171,60 +218,68 @@ export class BaseModel {
   public static async count<ModelType extends typeof BaseModel>(
     this: ModelType,
     filter: Filter<ModelAttributes<InstanceType<ModelType>>>,
-    options: CountDocumentsOptions = {},
+    options?: ModelAdapterOptions<CountDocumentsOptions>,
   ): Promise<number> {
     const collection = await this.getCollection();
-    return collection.countDocuments(filter, options);
+    const driverOptions = mergeDriverOptions(options);
+    return collection.countDocuments(filter, driverOptions);
   }
 
   public static async create<ModelType extends typeof BaseModel>(
     this: ModelType,
     value: Partial<ModelAttributes<InstanceType<ModelType>>>,
-    options?: InsertOneOptions,
+    options?: ModelAdapterOptions<InsertOneOptions>,
   ): Promise<InstanceType<ModelType>> {
     const collection = await this.getCollection();
+    const driverOptions = mergeDriverOptions(options);
     const instance = new this(value, {
       // @ts-expect-error Unavoidable error.
       collection,
-      session: options?.session,
+      session: driverOptions.session,
     }) as InstanceType<ModelType>;
-    await instance.save(options);
+    await instance.save({ driverOptions });
     return instance;
   }
 
   public static async createMany<ModelType extends typeof BaseModel>(
     this: ModelType,
     values: Array<Partial<ModelAttributes<InstanceType<ModelType>>>>,
-    options?: BulkWriteOptions,
+    options?: ModelAdapterOptions<BulkWriteOptions>,
   ): Promise<Array<InstanceType<ModelType>>> {
     const collection = await this.getCollection();
+    const driverOptions = mergeDriverOptions(options);
     const instances = values.map(
       (value) =>
         new this(value, {
           // @ts-expect-error Unavoidable error.
           collection,
-          session: options?.session,
+          session: driverOptions.session,
         }) as InstanceType<ModelType>,
     );
-    await Promise.all(instances.map((instance) => instance.save(options)));
+    for (const instance of instances) {
+      await instance.save({ driverOptions });
+    }
```

Comment:
This makes sure that the entries are created in the order that the user passed them.